payment/src/main/java/org/killbill/billing/payment/api/DefaultPaymentGatewayApi.java

```diff
@@ -103,6 +105,14 @@ public GatewayNotification doPaymentGatewayApiOperation(final Iterable<PluginPro
                                             final CallContext callContext,
                                             final WithPaymentControlCallback<T> callback) throws PaymentApiException {
 
+        final RequestData requestData = Request.getPerThreadRequestData();
+        final String requestId;
+        if (requestData != null) {
+            requestId = requestData.getRequestId();
+        } else {
+            requestId = "NotAvailableRequestId";
+        }
+
```

Comment:
We could use a one-liner here:
`final String requestId = Request.getPerThreadRequestData() != null ? Request.getPerThreadRequestData().getRequestId() : "NotAvailableRequestId";`

Also i am confused with the following:
1. Why don't we pass null (@Nullable) and do the mapping to `NotAvailableRequestId` at the time we need it?
2. Why do we need to get the requestId in this method if we don't use it here? Since this is a ThreadLocal we should extract the value where it is needed